R/write_validation_report.R

```diff
@@ -0,0 +1,94 @@
+#' Create validation report from template
+#' @param pkg_name name of package
+#' @param template what validation report template from {valtools} to use,
+#' passed to \code{usethis::use_template}
+#' @param open boolean whether to open the validation report, passed to \code{usetthis::use_template}
+#' @importFrom usethis use_template proj_set
+#' @importFrom tools file_ext
+#' @importFrom rlang with_interactive
+#' @export
+vt_use_report <- function(pkg_name = desc::desc_get_field("Package"),
+                          template = "validation_report.Rmd",
+                          open = FALSE){
+
+  tryCatch({
+    dir <- vt_find_config()
+  },
+  error = function(e){
+    abort(
+      paste0("Validation config file missing; instantiate via `vt_use_config()`."),
+      class = "vt.validation_config_missing"
+    )
+  })
+
+  val_leads <- get_val_leads(dir = dirname(dir))
+
+  if(length(val_leads) == 0){
+    val_leads <-  tryCatch({
+      val_leads_username <- with_interactive(value = FALSE, vt_username())
+      val_leads <- vt_get_user_info(username = val_leads_username,
+                                        type = "name")[["name"]]
+
+      config_details <- read_validation_config()
+      old_role <- config_details$usernames[[val_leads]]$role
+      config_details$usernames[[val_leads]]$role <-
+        ifelse(grepl(pattern = "validation lead", x = tolower(old_role)),
+               old_role,
+               paste0(c("Validation Lead", old_role ), collapse = ", "))
+      write_validation_config(path = dir,
+                              working_dir = config_details$working_dir,
+                              output_dir = config_details$output_dir,
+                              report_naming_format = config_details$report_naming_format,
+                              username_list = config_details$usernames)
+      return(val_leads)
+      }, error = function(e){
+        val_leads <- username(fallback = "")
+        vt_add_user_to_config(username = val_leads, name = val_leads, title = "",
+                              role = "Validation Lead")
+
+
+        inform(
+          message = paste0("User `", val_leads, "` does not exist in the config file.  ",
+                           "Assigning to role of `Validation Lead`"),
+          class = "vt.validation_config_missing_user_inform"
+        )
+        return(val_leads)
+      })
+  }
+
+
+  render_template( template = template,
+                output = file.path(get_config_working_dir(),
+                                   paste0(evaluate_filename(), ".", file_ext(template))),
```

Comment:
Are we expecting the validation.Rmd to take different names? evaluate_filename() is for use during knitting of the validation report.

Reply:
`validation_report.Rmd` is the template name (in valtools), but `evaluate_filename()` should pull the user-specified validation report name from `validation.yml` (in the package being validated) - I believe the default we suggest is "Validation_Report_{package}_v{version}_{date}". 

Is there a different way we should be getting the report name from validation.yml? 🤔 that the `evaluate_filename()` gives an algorithm in the case that users don't have config setup, but now I am thinking we might want to move that to an argument,...

Reply:
> user-specified validation report name

I take that to mean the _output_ from rendering validation report, since what does it matter what the original input name is, and then we need to have additional functions around identifying the starting validation report

Reply:
while I agree locking down the filename would make life easier for programming within {valtools}, I am wondering if we need the flexibility for SOP purposes? 

Also this breaks the default behavior that we generally expect from `rmarkdown::render`, not that we have to following existing defaults, but requiring the .Rmd to always be validation_report.Rmd is an opinionated implementation by {valtools}. For users who are creating the validation report .Rmd manually, how do we communicate that they have to name the .Rmd `validation_report.Rmd`?

Reply:
So far code has been written to assume the validation vignette is `validation.Rmd` - such that currently it is hard coded.

if we are to allow different validation vignette file names, here needs to be a separate validation_vignette field then in the config for valtools to find and handle it. currently the slot "report_naming_format" assumes this is the output naming format, so it includes information like date and version number. - I can make an update to the config to capture that info too

Reply:
see #123